### ISSUE TITLE
Recent buildtools update broke building corefx on Linux

### ISSUE THREAD

**Issue by akoeplinger** [dotnet/buildtools#197]
In https://github.com/dotnet/buildtools/commit/2343d6f36d96970b97cea6f210db90d54d2f7183 the Copy task in publishtest.targets was set to `UseHardlinksIfPossible="true"`. However, the xplat MSBuild doesn't implement hardlinks in a cross-platform way currently (it just pinvokes into CreateHardlink Windows API), so this crashes on !Windows.

We should probably check for the OS in publishtest.targets and try hardlinks only on Windows (since tests aren't run via msbuild on other platforms yet anyway).

/cc @JeremyKuhne @nguerrera

**Comment by akoeplinger** [comment]
I implemented hardlink support in xplat MSBuild with https://github.com/Microsoft/msbuild/pull/132.

**Comment by nguerrera** [comment]
@akoeplinger Thanks! Can we close this in that case?

**Comment by akoeplinger** [comment]
@nguerrera I'd prefer to keep this open until the issue is fixed, i.e. after my PR to msbuild is merged and an updated version used in buildtools here :)

**Comment by JeremyKuhne** [comment]
I don't expect we'll get an updated package any time soon- we should probably add the condition here pending that to unblock the build.

**Comment by akoeplinger** [comment]
Sure, whatever gets me a working buildtools is fine :)

**Comment by weshaggard** [comment]
@nguerrera Do we want to keep this issue open to track getting this re-enabled for x-plat builds?

**Comment by akoeplinger** [comment]
@weshaggard I filed a new issue tracking this: #209
